pkg/ucp/manifestservice/service.go

```
@@ -0,0 +1,126 @@
+
+// NewService creates a server to register manifests.
+func NewService(connection sdk.Connection, options ucpoptions.UCPConfig) *Service {
+	return &Service{
+		ucpConnection: connection,
+		options:       options,
+	}
+}
+
+// Name gets this service name.
+func (s *Service) Name() string {
+	return "manifestservice"
+}
+
+func waitForServer(ctx context.Context, host, port string, retryInterval time.Duration, timeout time.Duration) error {
+	address := net.JoinHostPort(host, port)
+	deadline := time.Now().Add(timeout)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("connection attempts canceled or timed out: %w", ctx.Err())
+		default:
+			conn, err := net.DialTimeout("tcp", address, retryInterval)
+			if err == nil {
+				conn.Close()
+				return nil
+			}
+
+			if time.Now().After(deadline) {
+				return fmt.Errorf("failed to connect to %s after %v: %w", address, timeout, err)
+			}
+
+			time.Sleep(retryInterval)
+		}
+	}
+}
+
+func (w *Service) Run(ctx context.Context) error {
+	logger := ucplog.FromContextOrDiscard(ctx)
+
+	// Set up signal handling for graceful shutdown
+	stopChan := make(chan os.Signal, 1)
+	signal.Notify(stopChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Start the manifest registration in a goroutine
+	//go func() {
+	// Define connection parameters
+	hostName := "localhost" //w.ucpConnection.Endpoint()/split to get host? // Replace with actual method
+	port := "9000"          // extract from endpoint Replace with actual method
+
+	// Attempt to connect to the server
+	err := waitForServer(ctx, hostName, port, 500*time.Millisecond, 5*time.Second)
+	if err != nil {
+		logger.Error(err, "Server is not available for manifest registration")
+		return nil
+	}
+
+	// Server is up, proceed to register manifests
+	manifestDir := w.options.Manifests.ManifestDirectory
+	if _, err := os.Stat(manifestDir); os.IsNotExist(err) {
+		logger.Error(err, "Manifest directory does not exist", "directory", manifestDir)
+		return nil
```

Return an error here instead of just logging. This should shut down the server.